### ISSUE TITLE
Add option for Solr pod affinity

### ISSUE THREAD

**Issue by timterle** [bloomberg/solr-operator#39]
**Is your feature request related to a problem? Please describe.**
If there are three nodes and SolrCloud "replicas" is set to 3, then all 3 Solr jvms are put on the same node. For performance and HA reasons it'd be preferred to spread out the Solr jvms.

**Describe the solution you'd like**
It should be possible to configure pod affinity so that the Solr jvms are spread on all available nodes. The zookeeper operator does this.

**Comment by HoustonPutman** [comment]
Hey @timterle , @sepulworld already has a PR in to add this functionality! Check out #35 . It's almost done but I think he might have had some more work to do on it before we can merge it.

**Comment by sepulworld** [comment]
Yes, sorry I was side tracked on some other things last couple weeks. Will get back to #35 soon!
